cachito/workers/nexus.py

```diff
@@ -149,3 +151,146 @@ def get_ca_cert():
     if config.cachito_nexus_ca_cert and os.path.exists(config.cachito_nexus_ca_cert):
         with open(config.cachito_nexus_ca_cert, "r") as f:
             return f.read()
+
+
+def get_component_info_from_nexus(
+    repository, component_format, name, version, group=None, max_attempts=1
+):
+    """
+    Get the component information from a Nexus repository using Nexus' REST API.
+
+    :param str repository: the name of the repository
+    :param str component_format: the format of the component (e.g. npm)
+    :param str name: the name of the component
+    :param str version: the version of the dependency; a wildcard can be specified but it should
+        not match more than a single version
+    :param str group: an optional group of the dependency (e.g. the scope of a npm package)
+    :param int max_attempts: the number of attempts to try to get a result; this defaults to ``1``
+    :return: the JSON about the component or None
+    :rtype: dict or None
+    :raise CachitoError: if the search fails or more than one component is returned
+    """
+    if max_attempts < 1:
+        raise ValueError("The max_attempts parameter must be at least 1")
+
+    component = None
+    attempts = 0
+    while component is None and attempts < max_attempts:
+        if attempts != 0:
+            log.warning(
+                "The component search did not yield any results. Trying again in three seconds."
+            )
+            time.sleep(3)
+
+        components = search_components(
+            format=component_format, group=group, name=name, repository=repository, version=version
+        )
+        if len(components) > 1:
+            raise CachitoError(
+                "The component search in Nexus unexpectedly returned more than result"
```

Comment:
"more than one result"

It would be nice to also log which components were found and the search parameters.